test/fail_compilation/test21930.d

```diff
@@ -0,0 +1,29 @@
+// https://issues.dlang.org/show_bug.cgi?id=21930
+/*
+TEST_OUTPUT:
+---
+fail_compilation/test21930.d(26): Error: function `D main` must return `int` or `void`
+fail_compilation/test21930.d(23): Error: variable `string` is used as a type
+fail_compilation/test21930.d(17):        variable `string` is declared here
+fail_compilation/test21930.d(28): Error: constructor `test21930.R.this(string)` is not callable using argument types `()`
+fail_compilation/test21930.d(19): Error: template instance `test21930.TP!(main)` error instantiating
```

Comment:
```suggestion
fail_compilation/test21930.d(21): Error: variable `string` is used as a type
fail_compilation/test21930.d(15):        variable `string` is declared here
fail_compilation/test21930.d(26): Error: constructor `test21930.R.this(string)` is not callable using argument types `()`
```
Inferred return type for `main` is no longer an error in `stable`

Reply:
You mean in `master` (as of https://github.com/dlang/dmd/pull/13195), right? Will fix soon.

Reply:
Indeed